packages/block-events-stream/src/blockRangeToLogs.ts

```diff
@@ -0,0 +1,52 @@
+import { EMPTY, OperatorFunction, concatMap, from, pipe, tap } from "rxjs";
+import { FetchLogsResult, fetchLogs } from "./fetchLogs";
+import { AbiEvent, Address } from "abitype";
+import { BlockNumber, PublicClient } from "viem";
+
+export type BlockRangeToLogsOptions<TAbiEvents extends readonly AbiEvent[]> = {
+  publicClient: PublicClient;
+  address?: Address | Address[];
+  events: TAbiEvents;
+  maxBlockRange?: bigint;
+};
+
+export type BlockRangeToLogsResult<TAbiEvents extends readonly AbiEvent[]> = OperatorFunction<
+  { startBlock: BlockNumber; endBlock: BlockNumber },
+  FetchLogsResult<TAbiEvents>
+>;
+
+export function blockRangeToLogs<TAbiEvents extends readonly AbiEvent[]>({
```

Comment:
should we add a code documentation comment above here that explains the purpose / behavior?